### PR TITLE
improve FIXME code

### PR DESCRIPTION
**Pull request by Kvicii**
Motivation:
improve some FIXME code

Modification:
1.for deprecated code, add `@deprecated` doc
2.improve FIXME code

### CODE REVIEW

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 1]
```diff
@@ -480,6 +480,7 @@ public boolean isDirectBufferPooled() {
     }
 
     /**
+     * @deprecated
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java:483]:
```suggestion
     * @deprecated will be removed
```

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 2]
```diff
@@ -489,6 +490,7 @@ public boolean hasThreadLocalCache() {
...
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java:493]:
```suggestion
     * @deprecated will be removed
```

**Review by normanmaurer — CHANGES_REQUESTED** [review]

**Comment by Kvicii** [comment]
@normanmaurer ok, thx.

**Comment by normanmaurer** [comment]
@Kvicii thanks

**Pull request merged**